unittests/Dialect/HW/InstanceGraphTest.cpp

```diff
@@ -37,4 +41,58 @@ TEST(InstanceGraphTest, PostOrderTraversal) {
   ASSERT_EQ(range.end(), it);
 }
 
+TEST(InstanceGraphCAPITest, PostOrderTraversal) {
+  MLIRContext context;
+
+  HWInstanceGraph instanceGraph =
+      hwInstanceGraphGet(wrap(fixtures::createModule(&context).getOperation()));
+
+  struct Context {
+    size_t i;
+    HWInstanceGraphNode topLevelNode;
+  };
+  auto ctx = Context{
+      .i = 0,
```

Comment:
@SpriteOvO FYI, I missed this in review, but this designated initializers syntax is a C++20 feature, but we're still on C++17 

https://github.com/llvm/circt/blob/e63a7e374b12e38a7cadacb88a011d661b47e2d1/CMakeLists.txt#L32

This doesn't seem to be checked by the compilers we use in pull request CI, but on main, the msvc compile failed (https://github.com/llvm/circt/actions/runs/7719228412/job/21042087110#step:10:4962).

I just sent a quick patch to change this here: https://github.com/llvm/circt/commit/e63a7e374b12e38a7cadacb88a011d661b47e2d1